### DISCUSSION TITLE
[JS101] 關於if/else statement

### DISCUSSION THREAD

**Discussion by ming320003** [Lidemy/forum#41] [前端相關問題]
老師好
這次想詢問的問題是在 5:36 提到的把 score>= 60 改為 score 也可以。因為重點是判斷它會不會是 true，如果是 true 就ok。又說會被當作false的值只有0、空字串、null。
之後老師寫了兩個程式來對照。
一個是
var score
if (score){
  console.log('pass')
}else{
  console.log('undefined')
}
答案是 undefined
但如果第一行的 var score 寫成 var score = 123 答案就會變成 pass

所以請問是因為 var score 的後面沒有接東西 (宣告了變數卻沒有給值 ) 所以 score 才變成 false 嗎？接了東西就會變成 true ？但是會被當作 false 的值不是只有0、空字串、null 嗎？

我想要知道答案所以自己做了一個實驗：
var score
console.log(score === true)
答案是 false，意思是有個變數 score，然後我想看看 score 本身到底是 true 還是 false？本來覺得
答案是 false 驗證了自已的猜測， 差別就在於後面有沒有接東西，如果沒有， score 就會像老師說的那樣會被當作 false 。(不過我有點疑惑的是，不是只有0、空字串、null 才是 false 嗎？是因為沒有接東西，所以就默認是 false？)
但是我找了找筆記好像沒有 score === true 的寫法，嘗試了
var score
console.log(score === true)
的答案也是 false ，所以我覺得應該是寫錯了。不過為甚麼不能直接 === true 呢？
謝謝老師

**Comment by aszx87410** [comment]
> 又說會被當作false的值只有0、空字串、null。

那應該是漏掉一個了喔，undefined 也會被當作 false

> 所以請問是因為 var score 的後面沒有接東西 (宣告了變數卻沒有給值 ) 所以 score 才變成 false 嗎？

是因為沒有接東西，預設值就會是 undefined

https://developer.mozilla.org/zh-CN/docs/Glossary/Falsy

**Reply by ming320003** [reply]
謝謝老師的解答
